### DISCUSSION TITLE
OAuth2.0 Redirect does not happen (Single-Sign-On)

### DISCUSSION THREAD

**Discussion by Trickfilm400** [Kong/insomnia#5799] [Help]
Hi,
I have a route on an express which is protected via OpenID Connect (so OAuth2)
The Express-Backend is using a session-based Single-Sign-On (The backend creates a Session after the SSO redirect back to the backend after the login on the SSO page)

In the webbrowser everything is working - I can login and have a session cookie and I am authenticated.

But with insomnia, I dont get a session cookie - the request to the backend just dont happen.
In the Response Timeline of the Access token, only the request to the token endpoint (`> POST /oauth/v2/token HTTP/2`) is mentioned.

But in the developer console there it says `Detected redirect https://callback-uri/?code=[...]` (which would be the correct way like in the webbrowser)
But this request does not happen, not in the timeline of the OAuth session nor am I able to find this in the backend

**So the question**:
- Why is this request back to the backend to the redirect uri not happening? Is this not the intended way of OAuth2 or is some configuration or anything else broken?
- -> Can Insomnia handle session-based OIDC requests?

Some details:
- used test example project for SSO: https://github.com/goranlisak/openid-connect-sample
- SSO Provider tried: Zitadel, Keycloak
- Also tried with postman - same issue

**Comment by KayleeTheMech** [comment]
Same Issue. Setup is Keycloak as identity provider and a scala backend. 

Keycloak's error is: "Cookie not found. Please make sure cookies are enabled in your browser." With a regular browser I'm able to access the API.
